### PR TITLE
Lesters/add local llms

### PR DESCRIPTION
**Pull request by lesters**
@bratseth Please review
@hmusum FYI
@arnej27959 FYI
@andreer FYI

This is **initial** support for running LLMs locally using llama.cpp through java-llama.cpp and the container-llama module. This supports most things such as concurrency and GPU support if shared libraries have been compiled with it. The `LocalLLMTest.java` can be run on a laptop and the included libraries support Mac GPUs. You just need to download a GGUF file for instance from HuggingFace.

The `llm-local-client.def` contains the most significant model parameters, but we need more experience with this to be able to tune this properly and run in the context of Vespa without blowing stuff up. Things like total memory use, GPU memory use, context window size, number of parallel completions and so on. I'll be adding some system and performance tests shortly to start this.

### CODE REVIEW

File: container-search/src/main/java/ai/vespa/llm/clients/LocalLLM.java
Comment (bratseth) [inline container-search/src/main/java/ai/vespa/llm/clients/LocalLLM.java:130]: Should we throw here?
Probably want to return a 429 and not fill the log with these ...

File: container-search/src/main/resources/configdefinitions/llm-local-client.def
Comment (bratseth) [inline container-search/src/main/resources/configdefinitions/llm-local-client.def:4]: Use a "model" parameter type instead of url to have the path, url and cloud options handled as one.

File: container-search/src/test/java/ai/vespa/llm/clients/LocalLLMTest.java
Comment (bratseth) [inline container-search/src/test/java/ai/vespa/llm/clients/LocalLLMTest.java:24]: Maybe we can find a super-small model that can be used for unit testing?

**Review by bratseth — COMMENTED** [review]
Great! See comments, also repeating my request to move the generic model stuff out of search.

**Comment by lesters** [comment]
@bratseth Found a relatively small LLM (1.1Mb) which is suitable for unit testing

**Comment by lesters** [comment]
Added DO NOT MERGE for some run-time testing of the dependencies.

@gjoranv Please take a look.

File: container-llama/src/main/java/de/kherud/llama/package-info.java
Comment (gjoranv) [inline container-llama/src/main/java/de/kherud/llama/package-info.java:6]: Do we really need to export this package?

**Review by gjoranv — DISMISSED** [review]
Please verify that vespa-yahoo builds before merging, to ensure that dependency enforcers don't fail.

**Comment by lesters** [comment]
@gjoranv Thanks for reminding me, have built vespa-yahoo and added dependencies there.

**Review by bratseth — APPROVED** [review]
Great!

**Pull request merged**
